src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -118,28 +122,106 @@ void AccountStoreSC<MAP>::InvokeInterpreter(
   }
 }
 
+// New Invoker for EVM returns the gas, maybe change this
+
+template <class MAP>
+uint64_t AccountStoreSC<MAP>::InvokeEvmInterpreter(
+    Account* contractAccount, INVOKE_TYPE invoke_type,
+    EvmCallParameters& params, const uint32_t& version, bool& ret,
+    TransactionReceipt& receipt) {
+  bool csUpdate{true};
+
+  evmproj::CallRespose evmReturnValues;
+  uint64_t gas = params.m_available_gas;
+
+  bool call_already_finished = false;
+
+  auto worker = [this, &params, &invoke_type, &ret, &receipt, &version,
+                 &call_already_finished, &evmReturnValues]() mutable -> void {
+    Json::Value jval;
+    if (invoke_type == RUNNER_CREATE || invoke_type == RUNNER_CALL) {
+      ret = EvmClient::GetInstance().CallRunner(
+          version, EvmUtils::GetEvmCallJson(params), evmReturnValues);
+    }
+
+    call_already_finished = true;
+    cv_callContract.notify_all();
+  };
+
+  DetachedFunction(1, worker);
+
+  {
+    std::unique_lock<std::mutex> lk(m_MutexCVCallContract);
+    if (!call_already_finished) {
+      cv_callContract.wait(lk);
+    } else {
+      LOG_GENERAL(INFO, "Call functions already finished!");
+    }
+  }
+
+  if (m_txnProcessTimeout) {
+    LOG_GENERAL(WARNING, "Txn processing timeout!");
+
+    EvmClient::GetInstance().CheckClient(0, true);
+    receipt.AddError(EXECUTE_CMD_TIMEOUT);
+    ret = false;
+  }
+
+  if (not evmReturnValues.isSuccess()) {
+    LOG_GENERAL(WARNING, evmReturnValues.ExitReason());
+  }
+
+  ret = evmReturnValues.isSuccess() ? ret : false;
+
+  if (evmReturnValues.Logs().size() > 0) {
+    LOG_GENERAL(WARNING, "Logs:" << evmReturnValues.Logs());
+    Json::Value v = "{ msg =\"" + evmReturnValues.Logs() + "\"" + "}";
+    receipt.AddException(v);
+  }
+
+  gas = EvmUtils::UpdateGasRemaining(receipt, invoke_type, gas,
+                                     evmReturnValues.Gas());
+
+  if (evmReturnValues.m_apply.OperationType() == "delete") {
+    this->RemoveAccount(Address(evmReturnValues.m_apply.Address()));
+  } else {
+    csUpdate = EvmUtils::EvmUpdateContractStateAndAccount(
```

Comment:
This also has nothing to do with the contractAccount.  You apply instructions to the address of every application. At this stage it doesn't matter what the current contract address is.